### PR TITLE
Add integration tests for TestExplorer

### PR DESCRIPTION
**Pull request by plemarquand**
Add some integration tests that exercise finding and running tests, ensuring they complete with the expected outcome.

This patch adds a new profile used when running the tests locally via the Extension Tests task. This ensures that the workspace is always in a clean and consistent state.

### CODE REVIEW

**Comment by plemarquand** [comment]
@swift-server-bot Test this please

**Comment by plemarquand** [comment]
@swift-server-bot Please test

**Comment by plemarquand** [comment]
@swift-server-bot Please test

**Comment by plemarquand** [comment]
@swift-server-bot Test this please

**Comment by plemarquand** [comment]
@swift-server-bot Test this please

**Review by adam-fowler — CHANGES_REQUESTED** [review]
For some reason the extension fails to load CodeLLDB when running tests. I get the error 
```
Error: Cannot activate the 'Swift' extension because it depends on unknown extension 'vadimcn.vscode-lldb'
```
Which then causes the TestDiscovery tests to hang.
If I run without the profile everything works

File: .vscode/testing-debug.code-profile
[the remark this answers is not in the document]
Comment (plemarquand) [reply .vscode/testing-debug.code-profile:1]: When you create a new profile these are what you get by default.

**Comment by plemarquand** [comment]
@adam-fowler I'm not sure why the CodeLLDB extension isn't present for you in the profile; the [.code-profile](https://github.com/swift-server/vscode-swift/pull/806/files#diff-3cdf9d533acf8ec324a2acec8acc09a5158e1532b463885603bab8c13faa690cR3) includes it. Do you have it installed in your regular VSCode/profile? I think inclusion in the .code-profile just enables it, but wont install it if it isn't present.

This is what I have in the extensions view in the test code profile:

Screenshot 2024-05-25 at 8 24 59 AM

**Comment by adam-fowler** [comment]
Worked out what the issue is. I didn't realise I have to import the profile before using it.
Currently I'm seeing a number of tests fail when running with Swift 5.10

**Comment by plemarquand** [comment]
@adam-fowler Which tests are failing specifically, and how are they failing? Do you have the logs?

**Comment by plemarquand** [comment]
@adam-fowler I've updated the CONTRIBUTING.md with instructions on importing the profile before running the tests.

**Review by adam-fowler — APPROVED** [review]
I think we're good

**Pull request merged**